lib/matplotlib/collections.py

```diff
@@ -549,11 +549,15 @@ def set_offsets(self, offsets):
         offsets : (N, 2) or (2,) array-like
         """
         offsets = np.asanyarray(offsets)
-        if offsets.shape == (2,):  # Broadcast (2,) -> (1, 2) but nothing else.
+        if offsets.shape == (2,):
             offsets = offsets[None, :]
-        self._offsets = np.column_stack(
-            (np.asarray(self.convert_xunits(offsets[:, 0]), 'float'),
-             np.asarray(self.convert_yunits(offsets[:, 1]), 'float')))
+
+        if offsets.shape == (2, 0):
```

Comment:
```suggestion
        if offsets.shape == (0, 2):
```

Which will require some changes to the tests.

The docs say(N, 2) so this should be the degenerate case of ((N, 2), .., (2, 2), (1, 2), (0, 2)) (the (2,) case is a special case for spelling (1, 2)).